sql/core/src/test/resources/sql-tests/inputs/extract.sql

```diff
@@ -83,3 +83,21 @@ select extract(epoch from c) from t;
 
 select extract(not_supported from c) from t;
 select extract(not_supported from i) from t;
+
+-- In SPARK-31476, we've supported extract('field', source), too
+select date_part('millennium', c) from t;
```

Comment:
I added only simple test patterns for `extract('field', source)`.

Reply:
I think only a couple of tests are fine :-).

Reply:
BTW, why is it `date_part` instead of `extract`? `date_part` seems being tested at `date_part.sql`.

Reply:
Wait .. are `DatePart` and `Extract` the same? If so, it might be better to have one class and handle aliases via `expression(..., setAlias=true)` at `FunctionRegistry`.

Reply:
Actually, this support (`extract('field', source)`) is a side-effect of #31476.... `EXTRACT` should be formed as `EXTRACT(field FROM source)`, so [the `example` field](https://github.com/apache/spark/blob/master/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/datetimeExpressions.scala#L2235-L2248) in `ExpressionDescription` is different now from each other. Related discussion: https://github.com/apache/spark/pull/21479#discussion_r409933251

Reply:
I see. +1 to have a separate examples and documentation.

Reply:
I think the side effect is we support `extract('field', source)`, why we are testing `date_part` here...

Reply:
Ur, my bad... I'll update soon...

Reply:
Updated.